pkg/debugapi/chequebook.go

```diff
@@ -48,3 +70,117 @@ func (s *server) chequebookAddressHandler(w http.ResponseWriter, r *http.Request
 	address := s.Chequebook.Address()
 	jsonhttp.OK(w, chequebookAddressResponse{Address: address.String()})
 }
+
+func (s *server) chequebookLastPeerHandler(w http.ResponseWriter, r *http.Request) {
+	addr := mux.Vars(r)["peer"]
+	peer, err := swarm.ParseHexAddress(addr)
+	if err != nil {
+		s.Logger.Debugf("debug api: chequebook lastcheque peer: invalid peer address %s: %v", addr, err)
+		s.Logger.Error("debug api: chequebook lastcheque peer: invalid peer address %s", addr)
+		jsonhttp.NotFound(w, errInvaliAddress)
+		return
+	}
+
+	lastSent, err := s.Swap.LastSentCheque(peer)
+	if err != nil && err != chequebook.ErrNoCheque {
+		s.Logger.Debugf("debug api: chequebook lastcheque peer: get peer %s last cheque: %v", peer.String(), err)
+		s.Logger.Errorf("debug api: chequebook lastcheque peer: can't get peer %s last cheque", peer.String())
+		jsonhttp.InternalServerError(w, errCantLastChequePeer)
+		return
+	}
+
+	lastReceived, err2 := s.Swap.LastReceivedCheque(peer)
+	if err2 != nil && err2 != chequebook.ErrNoCheque {
+		s.Logger.Debugf("debug api: chequebook lastcheque peer: get peer %s last cheque: %v", peer.String(), err2)
+		s.Logger.Errorf("debug api: chequebook lastcheque peer: can't get peer %s last cheque", peer.String())
+		jsonhttp.InternalServerError(w, errCantLastChequePeer)
+		return
+	}
+
+	var lastsent *chequebookLastChequePeerResponse
+	if err == nil {
+		lastsent = &chequebookLastChequePeerResponse{
+			Beneficiary: lastSent.Cheque.Beneficiary.String(),
+			Chequebook:  lastSent.Cheque.Chequebook.String(),
+			Payout:      lastSent.Cheque.CumulativePayout,
+		}
+	}
```

Comment:
Why not to put this before LastReceivedCheque call and avoid err2 and larger distance between handling these errors?

Reply:
refactored this partly with the order proposed here